blocks/components/editable/index.js

```diff
@@ -101,14 +110,26 @@ export default class Editable extends wp.element.Component {
 		this.props.onFocus();
 	}
 
-	onChange() {
-		if ( ! this.editor.isDirty() ) {
+	isActive() {
+		return tinymce.EditorManager.activeEditor.id === this.editor.id;
+	}
+
+	onSelectionChange() {
+		// We must check this because selectionChange is a global event.
+		if ( ! this.isActive() ) {
 			return;
 		}
 
-		this.savedContent = this.getContent();
-		this.editor.save();
-		this.props.onChange( this.savedContent );
+		this.content = this.getContent();
+		this.selection = this.getSelection();
+
+		if ( ! this.props.onChange ) {
+			return;
+		}
+
+		if ( ! isEqual( this.props.value, this.content ) ) {
```

Comment:
I see what you're getting at, but I also don't think it's desirable to undo one character at a time. We'll probably want to consider some interval to update state (e.g. debounced change). Do you have any insight on how browsers or TinyMCE manage it?